digdag-standards/src/main/java/io/digdag/standards/command/ecs/DefaultEcsClient.java

```diff
@@ -224,13 +249,61 @@ public GetLogEventsResult getLog(
         if (nextToken.isPresent()) {
             request.withNextToken("f/" + nextToken.get());
         }
-        return logs.getLogEvents(request);
+        return retryOnRateLimit(() -> logs.getLogEvents(request));
     }
 
     @Override
     public void close()
             throws IOException
     {
-        client.shutdown();
+        retryOnRateLimit(() -> {
+            client.shutdown();
+            return true; //avoid restrict on generics with void
```

Comment:
Returning null would be a bit better? I guess it'll be handled as T=Void in this case

Reply:
Thanks. Fixed it.